examples/demo/Advanced/Tabular_editor_with_context_menu_demo.py

```diff
@@ -0,0 +1,165 @@
+"""
+Defining column-specific context menu in a Tabular Editor. Shows how the
+example for the Table Editor can be adapted to work with a Tabular Editor.
+
+The demo is a simple baseball scoring system, which lists each player and
+their current batting statistics. After a given player has an at bat, you
+right-click on the table cell corresponding to the player and the result of
+the at-bat (e.g. 'S' = single) and select the 'Add' menu option to register
+that the player hit a single and update the player's overall statistics.
+
+This demo also illustrates the use of Property traits, and how using 'event'
+meta-data can simplify event handling by collapsing an event that can
+occur on a number of traits into a category of event, which can be handled by
+a single event handler defined for the category (in this case, the category
+is 'affects_average').
+"""
+
+from random import randint
+from traits.api import HasStrictTraits, Str, Int, Float, List, Property
+from traits.etsconfig.api import ETSConfig
+from traitsui.api import View, Item, TabularEditor
+from traitsui.menu import Menu, Action
+from traitsui.tabular_adapter import TabularAdapter
+
+
+# Define a custom tabular adapter for handling items which affect the player's
+# batting average:
+class PlayerAdapter(TabularAdapter):
+
+    alignment = 'center'
+
+    width = 0.09
+
+    def get_menu(self, object, trait, row, column):
+        column_name = self.column_map[column]
+        if column_name not in ['name', 'average']:
+            menu = Menu(Action(name='Add',
```

Comment:
From a style point of view, it may be better to have the menu be a module global or stored in a class trait, since there is no local context being used in the menu definitions.